File: sdks/php/src/Exceptions.php
<?php

declare(strict_types=1);

namespace Nexus\SDK;

use Exception;

/**
 * Base exception for Nexus SDK.
 */
class NexusException extends Exception
{
}

/**
 * Exception thrown when HTTP request fails.
 */
class NexusApiException extends NexusException
{
    public function __construct(
        public readonly int $statusCode,
        public readonly string $responseBody,
        string $message = ''
    ) {
        $msg = $message ?: "Nexus API error: HTTP {$statusCode}: {$responseBody}";
        parent::__construct($msg);
    }
}

/**
 * Exception thrown for transaction errors.
 */
class NexusTransactionException extends NexusException
{
}

File: sdks/csharp/Exceptions.cs
namespace Nexus.SDK;

/// <summary>
/// Base exception for Nexus SDK errors.
/// </summary>
public class NexusException : Exception
{
    public NexusException(string message) : base(message)
    {
    }

    public NexusException(string message, Exception innerException)
        : base(message, innerException)
    {
    }
}

/// <summary>
/// Exception thrown when an HTTP request to the Nexus API fails.
/// </summary>
public class NexusApiException : NexusException
{
    /// <summary>
    /// HTTP status code.
    /// </summary>
    public int StatusCode { get; }

    /// <summary>
    /// Response body.
    /// </summary>
    public string ResponseBody { get; }

    public NexusApiException(int statusCode, string responseBody)
        : base($"Nexus API error: HTTP {statusCode}: {responseBody}")
    {
        StatusCode = statusCode;
        ResponseBody = responseBody;
    }
}

/// <summary>
/// Exception thrown when a transaction operation fails.
/// </summary>
public class NexusTransactionException : NexusException
{
    public NexusTransactionException(string message) : base(message)
    {
    }

    public NexusTransactionException(string message, Exception innerException)
        : base(message, innerException)
    {
    }
}
